Cargo.toml

```diff
@@ -13,6 +13,7 @@ exclude = ["assets", "scripts", "benches"]
 [dependencies]
 num-traits = "0.2.18"
 multiversion = "0.7"
+num-complex = "0.4.5"
```

Comment:
I think this should be an optional feature, maybe even a non-default one but enabled on docs.rs. The associated functions then need to be annotated as available with this feature only; there is a rustdoc feature for that.

Reply:
I made the following changes:

- `num-complex` is now a optional and non-default feature
- associated functions and tests are annotated as available with this feature only
- doc will not creates docs for all features (not sure if that's the desired outcome here?)
- update the CI pipeline to run tests for all features (since I had to annotate tests that use `Complex`)